Lib/datetime.py

```diff
@@ -828,7 +828,7 @@ def replace(self, year=None, month=None, day=None):
             month = self._month
         if day is None:
             day = self._day
-        return date(year, month, day)
+        return type(self)(year, month, day)
```

Comment:
I prefer self.__class__, but I never know which one is the best :-)

Reply:
I don't have any real preference here. I usually go with `self.__class__` myself, but I was somewhat persuaded by the discussion [on StackOverflow](https://stackoverflow.com/a/10386227/467366) that seemed to indicate that they are equivalent in Python 3, and that `type` is shorter - seemed like a mixed bag as to which one is more intuitive. I'm happy to change it to the "house style".

Reply:
Not sure if it's even a relevant precedent, since they're in two different languages, but it does seem that the [C source code](https://github.com/python/cpython/blob/master/Modules/_datetimemodule.c#L5075) uses `Py_TYPE(self)` rather than some equivalent of `self.__class__`. Though for all I know that's a macro that expands to `self.__class__`...